app_dart/test/service/github_status_service_test.dart

```diff
@@ -0,0 +1,250 @@
+  ServiceAccountInfo serviceAccountInfo;
+  FakeConfig config;
+  MockBuildBucketClient mockBuildBucketClient;
+  LuciBuildService service;
+  GithubStatusService githubStatusService;
+  MockGitHubClient mockGitHubClient;
+  MockRepositoriesService mockRepositoriesService;
+
+  const Build macBuild = Build(
+    id: 999,
+    builderId: BuilderId(
+      project: 'flutter',
+      bucket: 'prod',
+      builder: 'MacDoesNotExit',
+    ),
+    status: Status.scheduled,
+  );
+
+  const Build linuxBuild = Build(
+    id: 998,
+    builderId: BuilderId(
+      project: 'flutter',
+      bucket: 'prod',
+      builder: 'Linux',
+    ),
+    status: Status.scheduled,
+  );
+
+  setUp(() {
+    serviceAccountInfo = const ServiceAccountInfo(email: 'abc@abcd.com');
+    config = FakeConfig(deviceLabServiceAccountValue: serviceAccountInfo);
+    mockBuildBucketClient = MockBuildBucketClient();
+    service =
+        LuciBuildService(config, mockBuildBucketClient, serviceAccountInfo);
+    githubStatusService = GithubStatusService(config, service);
+    mockGitHubClient = MockGitHubClient();
+    mockRepositoriesService = MockRepositoriesService();
+    when(mockGitHubClient.repositories).thenAnswer((_) {
+      return mockRepositoriesService;
+    });
+    config.githubClient = mockGitHubClient;
+  });
+  group('setBuildsPendingStatus', () {
+    test('Empty builds do nothing', () async {
+      when(mockBuildBucketClient.batch(any)).thenAnswer((_) async {
+        return const BatchResponse(
+          responses: <Response>[
+            Response(
+              searchBuilds: SearchBuildsResponse(
+                builds: <Build>[],
+              ),
+            ),
+          ],
+        );
+      });
+      await githubStatusService.setBuildsPendingStatus('flutter', 123, 'abc');
+      verifyNever(mockGitHubClient.repositories);
+    });
+
+    test('A build list create status', () async {
+      when(mockBuildBucketClient.batch(any)).thenAnswer((_) async {
+        return const BatchResponse(
+          responses: <Response>[
+            Response(
+              searchBuilds: SearchBuildsResponse(
+                builds: <Build>[linuxBuild],
+              ),
+            ),
+          ],
+        );
+      });
+      final List<RepositoryStatus> repositoryStatuses = <RepositoryStatus>[];
+      when(mockRepositoriesService.listStatuses(any, any)).thenAnswer((_) {
+        return Stream<RepositoryStatus>.fromIterable(repositoryStatuses);
+      });
+      await githubStatusService.setBuildsPendingStatus('flutter', 123, 'abc');
+      expect(
+          verify(mockRepositoriesService.createStatus(any, any, captureAny))
+              .captured
+              .first
+              .toJson(),
+          jsonDecode(
+              '{"state":"pending","target_url":"","description":"Flutter LUCI Build: Linux","context":"Linux"}'));
+    });
+  });
+  group('setPendingStatus', () {
+    setUp(() {});
+
+    test('Builder does not exist', () async {
```

Comment:
`Status not updated if builder does not exist`?

Reply:
Done